forgerock-core/src/main/java/org/forgerock/android/auth/RequestInterceptor.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth;
+
+import androidx.annotation.NonNull;
+
+/**
+ * Observes, modifies outgoing request from the SDK.
+ * Interceptors can be used to add, remove, or transform headers, url, method on the request.
+ */
+public interface RequestInterceptor {
+
+    /**
+     * Intercepts outgoing request from the SDK.
+     *
+     * @param request The original outgoing request
+     * @return The Updated Request
+     */
+    @NonNull
+    Request intercept(@NonNull Request request);
```

Comment:
I see a difference between the iOS and Android implementation here. iOS has Action as a second parameter. I believe it would also provide a better experience in Android, once the developer does not need to cast the 'tag' attribute from the request to determinate the type of action being intercepted.

Reply:
We have discussed this misalignment, seem iOS and JS have no strong opinion for the misalignment, as long as the intent are align. But to make it easier for developer to avoid the cast, I have created FRRequestInterceptor which has the same interface as other platforms. The tag is more generic, we could tag anything instead of bounding it to the Action object. I would like to keep the flexibility.

Reply:
Looks good!